### DISCUSSION TITLE
Laragon/windows with production target linux

### DISCUSSION THREAD

**Discussion by edgreenberg-mri** [leokhoa/laragon#643] [General]
Our production environment is LAMP.  PHP7.4 (going on 8), Mariadb 10.3, Redis and Centos (going to ubutnu). 

Development is done with a docker implementation of LAMP, running on windows. We use VSCode to edit the files. 

One of our developers suggests Laragon. If they want to use a Laragon implementation on Windows for their development, how much trouble might they have running the code on the LAMP environment?  Should I encourage or discourage them to experiment with this?

**Comment by leokhoa** [comment]
@edgreenberg-mri: 
I guess you/your team might encounter several challenges with Docker before considering Laragon and the associated question. 
Are there specific areas where you anticipate potential issues when the local environment is Windows/Laragon and the production environment is LAMP? 

There are a couple of primary concerns:

1.  Case sensitivity in Linux: The solution is to avoid hardcoding paths.
2. Versioning: Laragon facilitates easy matching of development and production environments.

In my view, bugs or issues are likely to arise at certain points. However, exercising caution and conducting thorough testing are crucial to prevent surprises when transitioning code from development to production. 

**Solution**: Establishing a staging environment.

 I believe you and your team will be happy with the stability of Laragon.
